.changeset/little-ads-return.md

```diff
@@ -0,0 +1,70 @@
+---
+"@makeswift/runtime": minor
+---
+
+BREAKING: The latest upgrade to the Makeswift client `getPages` method
+introduces sorting, path filtering, and pagination. This method was not
+previously paginated - in order to get all your pages, you may now use our
+`toArray` pagination helper method, which will automatically paginate through
+all results and aggregate them into an array:
+
+```tsx
+import { client } from '@/makeswift/client'
+import { MakeswiftPageResult } from '@makeswift/runtime/next'
+
+async function getAllPages(): Promise<MakeswiftPageResult[]> {
+  return await client.getPages().toArray()
+}
+```
+
+This change also deprecates the client `getSitemap` method, with the
+recommendation that sitemaps should now be generated using data returned from
+`getPages`. Note that the deprecation of `getSitemap` now involves the host
+being responsible for the construction of page paths in the sitemap (either with
+domain or path based localization). Below is an example that uses path-based
+localization with the `next-sitemap` library:
```

Comment:
Should we show the example using App Router and its native support for sitemaps?

Reply:
I was planning on documenting this in our actual documentation page, as opposed to here (which is more focused on the deprecation). Do we need to do both?

Reply:
I think our release and our upgrade guide should be similar and/or link to each other. Also, the App Router example might be cleaner since it doesn't involve external dependencies, etc.